edx_solutions_api_integration/courses/views.py

```diff
@@ -1930,6 +1922,107 @@ def get(self, request, course_id):  # pylint: disable=W0613,W0221
+        }
+
+    Filters can also be applied
+    ```/api/courses/{course_id}/completions/?user_id={user_id}```
+    ```/api/courses/{course_id}/completions/?content_id={content_id}```
+    ```/api/courses/{course_id}/completions/?user_id={user_id}&content_id={content_id}```
+    - POST: Creates a Course-Module completion entity
+    - POST Example:
+        {
+            "content_id":"i4x://the/content/location",
+            "user_id":4,
+            "stage": "First"
+        }
+    ### Use Cases/Notes:
+    * Use GET operation to retrieve list of course completions by user
+    * Use GET operation to verify user has completed specific course module
+    """
+    serializer_class = BlockCompletionSerializer
+
+    def get_queryset(self):
+        """
+        GET /api/courses/{course_id}/completions/
+        """
+        content_id = self.request.query_params.get('content_id', None)
+        course_id = self.kwargs['course_id']
+        if not course_exists(self.request, self.request.user, course_id):
+            raise Http404
+        course_key = get_course_key(course_id)
+        queryset = BlockCompletion.objects.filter(course_key=course_key).select_related('user')
+        user_ids = get_ids_from_list_param(self.request, 'user_id')
+        if user_ids:
+            queryset = queryset.filter(user__in=user_ids)
+
+        if content_id:
+            content_descriptor, content_key, _existing_content = get_course_child(
+                self.request,
+                self.request.user,
+                course_key,
+                content_id
+            )
+            if not content_descriptor:
+                raise Http404
+            queryset = queryset.filter(block_key=content_key)
+
+        return queryset
+
+    def post(self, request, course_id):
+        """
+        POST /api/courses/{course_id}/completions/
+        """
+        warnings.warn(DeprecationWarning, "BlockCompletions should not be created this way.  Use the suggested APIs instead")
+        content_id = request.data.get('content_id', None)
+        user_id = request.data.get('user_id', None)
+        stage = request.data.get('stage', None)
+        if not content_id:
+            return Response({'message': _('content_id is missing')}, status.HTTP_400_BAD_REQUEST)
+        if not user_id:
+            return Response({'message': _('user_id is missing')}, status.HTTP_400_BAD_REQUEST)
+        if not course_exists(request, request.user, course_id):
+            return Response({}, status=status.HTTP_404_NOT_FOUND)
+        course_key = get_course_key(course_id)
+        content_descriptor, content_key, existing_content = get_course_child(request, request.user, course_key, content_id)  # pylint: disable=W0612,C0301
+        if not content_descriptor:
+            return Response({'message': _('content_id is invalid')}, status.HTTP_400_BAD_REQUEST)
+
+        completion, created = BlockCompletion.objects.submit_completion(
+            user_id=user_id,
```

Comment:
@jcdyer, there is no 'user_id` keyword argument for the `submit_completion` method. There is a `user` keyword argument which accepts a `User` object and not a `user_id`.